pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -484,6 +484,18 @@ default boolean defaultInferPartitionHint() {
       return CommonConstants.Broker.DEFAULT_INFER_PARTITION_HINT;
     }
 
+    /**
+     * Whether to use spools or not.
+     *
+     * This is treated as the default value for the broker and it is expected to be obtained from a Pinot configuration.
+     * This default value can be always overridden at query level by the query option
+     * {@link CommonConstants.Broker.Request.QueryOptionKey#USE_SPOOLS}.
+     */
+    @Value.Default
+    default boolean defaultUseSpools() {
+      return CommonConstants.Broker.DEFAULT_OF_SPOOLS;
+    }
```

Comment:
It is configurable. This is just the default value in case `defaultUseSpools(boolean)` is not called (which is called in `MultiStageBrokerRequestHandler`). It is following the same design principles as `defaultInferPartitionHint` above. See https://immutables.github.io/immutable.html#default-attributes

Reply:
I see, thanks for the pointer.